src/System.Collections.Immutable/tests/ImmutableListBuilderTest.cs

```diff
@@ -178,6 +178,17 @@ public void Remove()
         }
 
         [Fact]
+        public void RemoveAllBugTest()
+        {
+            var builder = ImmutableList.CreateBuilder<int>();
+            var elemsToRemove = new[]{0, 1, 2, 3, 4, 5}.ToImmutableHashSet();
+            foreach(var elem in new[]{0, 1, 2, 3, 4, 5, 6})
+                builder.Add(elem);
```

Comment:
On further investigation, AddRange does not exhibit the issue.  I suspect it is related to how the tree gets built up in that case.  I suppose I could figure out the right incantations to do this correctly, or add a comment that indicates why Add is used.  I would hate for someone to "optimize" this later and end up with a dud test.